sys/userlib/src/kipc.rs

```diff
@@ -162,3 +162,37 @@ pub fn software_irq(task: usize, mask: u32) {
         &[],
     );
 }
+
+/// Reads a task's panic message into the provided `buf`, if the task is
+/// panicked.
+///
+/// # Returns
+///
+/// - [`Ok`]`(&[u8])` if the task is panicked. The returned slice is borrowed
+///    from `buf`, and contains the task's panic message as a sequence of
+///   UTF-8 bytes. Note that the slice may be empty, if the task has panicked
+///   but was compiled without panic messages enabled.
+/// - [`Err`]`(`[`ReadPanicMessageError::TaskNotPanicked`]`)` if the task is
+///   not currently faulted due to a panic.
+/// - [`Err`]`(`[`ReadPanicMessageError::BadPanicMessage`]`)` if the task has
+///   panicked but the panic message buffer is invalid to read from.
+pub fn read_panic_message(
+    task: usize,
+    buf: &mut [u8; 128],
```

Comment:
Promote `const BUFSIZE: usize = 128;` out of the panic handler into a global constant (`PANIC_BUFSIZE`), then use it here?  It could even be public so others can size their buffers appropriately.